tests/frontier/precompiles/test_precompiles.py

```diff
@@ -109,3 +115,107 @@ def test_precompiles(
 
     state_test(env=env, pre=pre, post=post, tx=tx)
+
+
+def precompile_addresses_in_successor(fork: Fork) -> Iterator[Tuple[str, bool]]:
+    """
+    Yield the addresses of precompiled contracts and whether they existed in the parent fork.
+
+    Args:
+        fork (Fork): The transition fork instance containing precompiled contract information.
+
+    Yields:
+        Iterator[Tuple[str, bool]]: A tuple containing the address in hexadecimal format and a
+            boolean indicating whether the address has existed in the predecessor.
+
+    """
+    predecessor_precompiles = [
+        hex(int.from_bytes(address, byteorder="big"))
+        for address in get_transition_fork_predecessor(fork).precompiles()
+    ]
+    successor_precompiles = [
+        hex(int.from_bytes(address, byteorder="big"))
+        for address in get_transition_fork_successor(fork).precompiles()
+    ]
+    for address in successor_precompiles:
+        if address in predecessor_precompiles:
+            yield address, True
+        else:
+            yield address, False
+
+
+@pytest.mark.valid_at_transition_to("Paris", subsequent_forks=True)
+@pytest.mark.parametrize_by_fork(
+    "address,exists_in_predecessor", precompile_addresses_in_successor
+)
+def test_precompile_warming(
+    blockchain_test: BlockchainTestFiller, address: str, exists_in_predecessor: bool, pre: Alloc
+):
+    """
+    Call BALANCE of a precompile addresses before and after a fork.
+
+    According to EIP-2929, when a transaction begins, accessed_addresses is initialized to include:
+    - tx.sender, tx.to
+    - and the set of all precompiles
+
+    This test verifies that:
+    1. Precompiles that exist in the predecessor fork are always "warm" (lower gas cost)
+    2. New precompiles added in a fork are "cold" before the fork and become "warm" after
+
+    """
+    sender = pre.fund_eoa()
+    call_cost_slot = 0
+
+    code = (
+        Op.GAS
+        + Op.BALANCE(address)
+        + Op.POP
+        + Op.SSTORE(call_cost_slot, Op.SUB(Op.SWAP1, Op.GAS))
+        + Op.STOP
+    )
+    before = pre.deploy_contract(code, storage={0: 0xDEADBEEF})
+    after = pre.deploy_contract(code, storage={0: 0xDEADBEEF})
+
+    # Block before fork
+    blocks = [
+        Block(
+            timestamp=10000,
+            txs=[
+                Transaction(
+                    sender=sender,
+                    to=before,
+                    gas_limit=1_000_000,
+                )
+            ],
+        )
+    ]
+
+    # Block after fork
+    blocks += [
+        Block(
+            timestamp=20000,
+            txs=[
+                Transaction(
+                    sender=sender,
+                    to=after,
+                    gas_limit=1_000_000,
+                )
+            ],
+        )
+    ]
+
+    if not exists_in_predecessor:
+        expected_gas_before = 2607
+    else:
+        expected_gas_before = 107
```

Comment:
Can be programmatic with `gas_costs = fork.gas_costs()`, and then the warm cost is `gas_costs.G_WARM_ACCOUNT_ACCESS` and the cold cost is `gas_costs.G_COLD_ACCOUNT_ACCESS`.

The extra 7 is Op.GAS (gas_costs.G_BASE), Op.PUSHx (gas_costs.G_VERY_LOW), and Op.POP (gas_costs.G_BASE).